specs/core/0_beacon-chain.md

```diff
@@ -1596,41 +1611,35 @@ _Note_: If there are skipped slots between a block and its parent block, run the
 
 ### Per-slot processing
 
-Below are the processing steps that happen at every slot.
-
-#### Slot
+Below are the processing steps that happen at every `slot >= GENESIS_SLOT`.
 
+* Set `state.latest_state_roots[state.slot % SLOTS_PER_BATCHING] = hash_tree_root(state)`.
 * Set `state.slot += 1`.
-
-#### Block roots
-
-* Let `previous_block_root` be the `hash_tree_root` of the previous beacon block processed in the chain.
-* Set `state.latest_block_roots[(state.slot - 1) % LATEST_BLOCK_ROOTS_LENGTH] = previous_block_root`.
-* If `state.slot % LATEST_BLOCK_ROOTS_LENGTH == 0` append `merkle_root(state.latest_block_roots)` to `state.batched_block_roots`.
+* Set `state.latest_block_roots[state.slot % SLOTS_PER_BATCHING] = get_block_root(state, state.slot - 1)`.
```

Comment:
These are really `block_body_roots` now. I think we should reserve `block_root` for the actual root of the block.

I'm a bit concerned about the mismatch between storing block body roots in the state vs actual block roots which will likely be used on the p2p network.

Also this changes what committees are attesting to. They used to attest to the actual block root. Now the validity rule is about attesting to this body (which doesn't even include the state/header). We are certainly changing meaning here and need to make sure it's actually the intention.

Reply:
Also due to lines like "Verify that `attestation.data.justified_block_root` is equal to `get_block_root(state, get_epoch_start_slot(attestation.data.justified_epoch))`", we are really just finalizing the block body which isn't even a commitment to a chain (no `parent_root` in block body) and we could see the same block body in multiple chains.

Reply:
These are all great points! We discussed and agreed to rollback to block roots with the following changes:

* Add `latest_block_body_root` so that `parent_block_root` can be reconstructed from `state`.
* Define `block_root` to be `signed_root(block.header, "signature")`.

One discussion point is that we need canonical identifiers for various signed objects (e.g. blocks, attestations, transfers), for example in the context of block explorers. Our possibility is to use `signed_root(object, "signature")` as that canonical identifier.

Reply:
I think defining `block_root = signed_root(block.header, "signature")` isn't ideal so I've reverted back to `block_root = hash_tree_root(block.header)`. Technically it's a `block_header_root` though...

Reply:
Ok no `block_root = hash_tree_root(block)` and order is restored 👍 :)